sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -400,54 +309,62 @@ static az_result waiting(az_event_policy* me, az_event event)
       // No-op
       break;
 
+    case AZ_MQTT5_EVENT_SUBACK_RSP:
+    {
+      // if get suback that matches the sub we sent, stop waiting for the suback
+      az_mqtt5_suback_data* data = (az_mqtt5_suback_data*)event.data;
+      if (data->id
+          == this_policy->_internal.rpc_server_memory._internal._az_mqtt5_rpc_server_pending_sub_id)
```

Comment:
The long full-struct name here is more reason to simply attach the contents of the "memory" directly to the state machine's object (i.e. this_policy->_internal.subscription_id).

Also - the name of the internal shouldn't be this long (instead of `_az_mqtt5_rpc_server_pending_sub_id` it could be just `id` or `subscription_packet_id`).